Sources/CRC/CRC32.swift

```diff
@@ -40,16 +40,90 @@ struct CRC32:Hashable, Sendable
         return checksum
     }
 
+    /// Returns a new checksum by hashing the provided message into the current checksum.
+    ///
+    /// This manually specialized implementation is much faster in debug mode than the
+    /// generic implementation, but exactly the same in release mode.
+    @inlinable public
```

Comment:
should this be `consuming`, and should there also be a generic `Sequence<UInt8>` overload?

Reply:
There's already a generic overload above (which this manually specified method is kinda a copy of). I don't think `consuming` would make a difference to debug mode performance, but I do agree that it'd make sense for both the existing method and this method to have.

Reply:
this type is bitwise copyable, so `consuming` would not affect performance, it is more a matter of “usage hint”. it’s also a bit simpler to spell the body of the function, as you simply mutate `self` and return it

please do add a `consuming func updated(with message:borrowing some Sequence<UInt8>)`, somebody (probably myself) will inevitably reach for it

also, since these are performance hooks, i would *really* prefer if the `[UInt8]` overloads were prefixed with an underscore. overloading *sucks* in Swift, and it would be nice if documentation authors (both in this package and downstream) could still refer to `update(with:)` without having to guess an FNV-1 hash :)